### DISCUSSION TITLE
How to clone a vertex with edges

### DISCUSSION THREAD

**Discussion by rautjris** [Gremlinq/ExRam.Gremlinq#552] [Q&A]
Hi !

Need some advice how to clone a Vertex with it's properties and Edges/edge properties to a new Vertex.
I'm newbie with gremlin and using Cosmos Db gremlin.

This is what I have now:

await _g.V<SampleVertex>(sourceid).As((_, source) => _
 .AddV(new SampleVertex())
 .As(( _ , clone) => _
 .SideEffect(s =>
 {
 return s.Select(source).OutE<SampleEdge>().As(( _ , e) => _
 .Select(e).InV().AddE<SampleEdge>().From(f => f.Select(clone)));
 }
 )
));

But I'm out of ideas how to copy properties.

Thanks in advance

**Comment by danielcweber** [comment]
See #553.
